engine/common/requester/engine.go

```diff
@@ -306,29 +306,17 @@ func (e *Engine) dispatchRequest() (bool, error) {
 	e.log.Debug().Int("num_entities", len(e.items)).Msg("selecting entities")
 
 	// get the current top-level set of valid providers
-	providers, err := e.state.Final().Identities(e.selector)
+	final := e.state.Final()
+	providers, err := final.Identities(e.selector)
 	if err != nil {
 		return false, fmt.Errorf("could not get providers: %w", err)
 	}
 
-	// randomize order of items, so that they can be requested in different order each time
-	rndItems := make([]flow.Identifier, 0, len(e.items))
```

Comment:
The e.items is a map, iterating it is already close to randomly iterating. We don't need to explicitly shuffle the items here. And in fact, doing this would increase time complexity to O(len(e.items)). Removing this random shuffle, could reduce the time complexity to O(max(BatchThreshold, len(e.items))).